uniffi_bindgen/src/interface/object.rs

```diff
@@ -313,22 +321,29 @@ impl Method {
 
     pub fn first_argument(&self) -> Argument {
         Argument {
-            name: "handle".to_string(),
+            name: "ptr".to_string(),
             // TODO: ideally we'd get this via `ci.resolve_type_expression` so that it
             // is contained in the proper `TypeUniverse`, but this works for now.
             type_: Type::Object(self.object_name.clone()),
-            by_ref: false,
+            by_ref: !self.attributes.get_self_by_arc(),
             optional: false,
             default: None,
         }
     }
 
+    pub fn all_arguments(&self) -> Vec<Argument> {
+        vec![self.first_argument()]
+            .into_iter()
+            .chain(self.arguments.iter().cloned())
+            .collect()
+    }
```

Comment:
So here for methods, we're making a distinction between the "visible to the user" `arguments()` list and the "what you see in the actual call" `all_arguments()` list...

Reply:
In fact, `first_argument` can be removed completely, because we only ever used it in order to build up the list that is now returned by `all_arguments`.